src/Microsoft.Sbom.Api/Config/ConfigSanitizer.cs

```diff
@@ -45,6 +45,12 @@ public IConfiguration SanitizeConfig(IConfiguration configuration)
                 .WriteTo.Console(outputTemplate: Constants.LoggerTemplate)
                 .CreateLogger();
 
+            // If BuildDropPath is null then run the logic to check whether it is required or not based on the current configuration.
+            if (configuration.BuildDropPath?.Value == null || (configuration.DockerImagesToScan?.Value != null && configuration.BuildComponentPath?.Value == null))
```

Comment:
I don't think configuration.BuildComponentPath?.Value == null check is needed since it was true, the second check will never be reached

Reply:
If configuration.BuildDropPath?.Value == null and configuration.DockerImagesToScan?.Value == null, this should throw, right?

Reply:
This check is there to cover the scenario where the user provides -b and -di but no -bc and no -m.

![bAndDi](https://user-images.githubusercontent.com/69322674/232842719-8570bb08-de7e-46be-ae06-a75024378341.png)

Reply:
Correct, If configuration.BuildDropPath?.Value == null and configuration.DockerImagesToScan?.Value == null we will throw an error with a message asking for a BuildDropPath.